src/EditorFeatures/Test2/IntelliSense/CompletionServiceTests.vb

```diff
@@ -80,5 +82,60 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
                 Return Task.CompletedTask
             End Function
         End Class
+
+        <Fact>
+        Public Async Function TestProviderForDifferentTextViewRoles() As Task
+            Dim workspaceDefinition =
+            <Workspace>
+                <Project Language="C#" AssemblyName="TestAssembly" CommonReferencesPortable="true">
+                    <Document>
+$$
+                </Document>
+                </Project>
+            </Workspace>
+
+            Dim composition = EditorTestCompositions.EditorFeatures.AddParts(GetType(MyRoleProvider))
+
+            Using workspace = TestWorkspace.Create(workspaceDefinition, composition:=composition)
+                Dim document = workspace.CurrentSolution.Projects.First.Documents.First
+                Dim completionService = document.GetRequiredLanguageService(Of CompletionService)()
+
+                Dim list = Await completionService.GetCompletionsAsync(
+                    document, caretPosition:=0, CompletionOptions.Default, OptionValueSet.Empty, CompletionTrigger.Invoke,
+                    roles:=ImmutableHashSet.Create("MyTextViewRole"))
+
+                Assert.True(list.ItemsList.Contains(MyRoleProvider.Item))
+
+                Dim myRoleDescription = Await completionService.GetDescriptionAsync(document, MyRoleProvider.Item, CompletionOptions.Default, SymbolDescriptionOptions.Default)
+                Assert.Equal(MyRoleProvider.DescriptionText, myRoleDescription.Text)
+            End Using
+        End Function
+
+        <ExportCompletionProviderMef1(NameOf(MyRoleProvider), LanguageNames.CSharp)>
+        <Microsoft.VisualStudio.Text.Editor.TextViewRole("MyTextViewRole")>
```

Comment:
I tried to test this with real `CSharpInteractiveWindowCommandCompletionProvider`, but it's hardcoded to use `InteractiveWindowWorkspace` and I couldn't find an easy way to hook it up in unit tests